packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -610,22 +661,41 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
             }
 
             if (!this.notBoundedChannelContextSet.has(objectId)) {
-                let summary: ISummaryTreeWithStats;
+                let summaryTree: ISummaryTreeWithStats;
                 if (value.isLoaded) {
-                    summary = value.getAttachSummary();
+                    const summarizeResult = value.getAttachSummary();
+                    assert(
+                        summarizeResult.summary.type === SummaryType.Tree,
+                        "getAttachSummary should always return a tree");
+                    summaryTree = { stats: summarizeResult.stats, summary: summarizeResult.summary };
+
+                    // Add the context's Fluid object references to the list of this channel's references.
+                    references.push(...summarizeResult.references);
                 } else {
                     // If this channel is not yet loaded, then there should be no changes in the snapshot from which
                     // it was created as it is detached container. So just use the previous snapshot.
                     assert(!!this.dataStoreContext.baseSnapshot,
                         "BaseSnapshot should be there as detached container loaded from snapshot");
-                    summary = convertSnapshotTreeToSummaryTree(this.dataStoreContext.baseSnapshot.trees[objectId]);
+                    summaryTree = convertSnapshotTreeToSummaryTree(this.dataStoreContext.baseSnapshot.trees[objectId]);
                 }
-
-                builder.addWithStats(objectId, summary);
+                builder.addWithStats(objectId, summaryTree);
             }
         }
 
-        return builder.getSummaryTree();
+        // Normalize all the context's path of Fluid object references and prefix them with this channel's id.
+        normalizeAndPrefixReferencesPath(references, this.id);
```

Comment:
can we please reduce code duplication? And reuse code between these two summarization methods?